src/mono/mono/metadata/marshal.c

```diff
@@ -6264,6 +6264,31 @@ mono_marshal_asany_impl (MonoObjectHandle o, MonoMarshalNative string_encoding,
 
 		return res;
 	}
+	case MONO_TYPE_SZARRAY: {
+		MonoClass *klass = t->data.klass;
+		MonoClass *eklass = m_class_get_element_class (klass);
+		MonoArray *arr = (MonoArray *) MONO_HANDLE_RAW (o);
+
+		if (m_class_get_rank (klass) > 1)
+			break;
+
+		if (arr->bounds)
+			if (arr->bounds->lower_bound != 0)
```

Comment:
Is this a permanent restriction, or just one you're putting in place for now? Not sure how CoreCLR handles non-0 lower bounds.

Reply:
CoreCLR does the same check I believe, and there is no concept of lower bound in native arrays.

Reply:
Sounds good. And yeah, but you could ignore the bounds or pad with NULL bytes, so just wanted to make sure we're the same as CoreCLR.